tardis/visualization/tools/sdec_plot.py

```diff
@@ -1178,6 +1181,33 @@ def generate_plot_mpl(
                 linewidth=1,
             )
 
+        # Plot observed spectrum
+        if observed_spectrum:
+            observed_spectrum_wavelength = None
+            observed_spectrum_luminosity = None
+
+            if (
+                type(observed_spectrum[0]) is u.quantity.Quantity
+                and type(observed_spectrum[1]) is u.quantity.Quantity
+            ):
+                # If the observed spectrum is a list of quantities, then convert to wavelength and luminosity units
+                observed_spectrum_wavelength = observed_spectrum[0].to(u.AA)
+                observed_spectrum_luminosity = observed_spectrum[1].to(
+                     "erg / s"
+                )
+
+            else:
+                raise ValueError(
+                    "Both wavelength and luminosity should be astropy quantities."
+                )
+            self.ax.plot(
+                observed_spectrum_wavelength,
+                observed_spectrum_luminosity/ self.lum_to_flux,
```

Comment:
I think observed spectrum's y passed by user will be flux (not luminosity that we need to convert), isn't so @jamesgillanders @MarkMageeAstro?

IF yes, @atharva-2001 an exception must be raised if user passed observed_spectrum but didn't pass distance

Reply:
yes this is true - in most instances, the user will be passing an observed spectrum in flux